### ISSUE TITLE
Base64 encoded Japanese character not decoded as expected

### ISSUE THREAD

**Issue by kztmk** [emailjs/emailjs-mime-parser#35]
Hi, I tried an email via emailjs-imap-client and parse it. but Japanese charctors were broken only base64 encoded mail. If you need some more test data, I can upload to my server.

Please check it.

Thank you.

## Expected Behavior
[http://test.serve2you.net/images/expected.png](http://test.serve2you.net/images/expected.png)

## Current Behavior
[http://test.serve2you.net/images/result.png](http://test.serve2you.net/images/result.png)

## raw data in MimeNode
"X-Apparently-To: okb2qjat@yahoo.co.jp via 183.79.150.63; Sun, 29 Apr 2018 15:10:58 +0900
X-YMailISG: LLBZCYkWLDsxwFGjqC2NXRY69rlxy_85oO_xzUMpyyyNugB3
 Gd6GV1qGI8lwfuVh6wG1gjCiyb2WjOFm.92tG3oC7drcMZmtGr4bQzLV2kDm
 0PzmnhipOi7gF1QxcfoEFcvNjLODLys1cbJmX.fzf3C5Y1vc1DkfKvNibnfy
 mpOe0y5isP_GsIpAiot1FcIga_2n2vL6Xtyu52okfUGOAWa1NwfQspHgvK_L
 ozZygxTHIv0zceDH.RoApm7hBQ4Erw_pPzGbP_IkuRD.IzlBWLdg31D5PhFb
 zIfnnJDaxsu8b5FsVSJvWwlcwrvm8cAn8RVzLljQmun16GdAVhd0f5D1p1vn
 nLgGKtPPMZtPpXjd.clC.fM4y_Nt_U3afDrVKEizjVejKXSOtbpaj9ObD5Cx
 UYz3c4TV_H3m2j_CB1WFKggrVWyUwjCWxAbzZ7AUYgTUMcXkxZx41UNjVBpH
 e_6lersBeoK59SR4tCQqkVXZ9dl6TrnVgRX2aZjERpSxlgzgXOrBiB2WL52e
 V69zlIOYHdwkAfdhM7Sbdsxis4bKzxgZIYoSFIbq.BAgwkRgJJVzb2mtAMD8
 agieIHMfg.R7nKAMEAuRNynRS4LcaVug0Nxrqjbt9bRhj.g5FTZGnt8wECCu
 FTbG7vtka7ziOn0w55Rq7jyaptXSMb5LJcuw6RgeyQL5WNjTdZ1BswLj.y2p
 TGj19PSFduxQ5lwid4MT9op1Ma.qXDY9Y5oWytfssRvTN57T_LGCRnunGEF7
 sG_JZSpMBN_MXHeLItp1HRrQkwjIEsjHdWB3hDqy32ULwEOFvmXhZjZX033j
 8vU1k2OtyKU2Yf34MnwbnANj9CHASH6LQQt6z45QZVD6GdlJRV68edxrD_yg
 L0TA6KCFfaNCqBzb1Ug3gmddzf6WLu7cekH98QdMxGifL0AAuEtvZEFZiDQy
 ceYJh8T5mFToVmktwaBHgx0WTrjclne6GB9dxnmgHLVYFDO32f3yemu0Fid.
 ZBB7hc3dp5ardrMurwmU5iBdpvL_Y84cdhOQPO67oag8oH1CDQiYYPx7HcZq
 _M9TAX6kS2Z_5UY3hfpk8hJExTm4nth1R9eBtQLwwejK1m0EndCJ3H1SdBHt
 n.2X5yEGu_BmAaf5RwxMn42SH3srrqNTb6ziHm_vNC37RF11ACzCb8f.Vjot
 PSHpx4BPX40UyDsz.Qc6uHnIK1n8TTKlYAW24Xjc22T6m.X_tuXVO3KzsUw8
 n0M2ffRQh9EGAlgpSHAa0arUaqpSJWcfYDl1bVQT0F9Okaz67lPdL5kNQUSy
 p_TwnZoT.5Ryv4ekf.0x7RPHthG6HPhd0l62hflioBHM5QZUWCLWXL1yrz40
 tjmLagBKHGTWZfusTsu1yEdK_ECwyzGxaMdg4Rb_qIjJuWW7rCzt_2gJwBVU
 EaMB1fPA8LWjPROm_WGiyV1F8QSkxKCASaDbCOsg8KU-
X-Originating-IP: [183.79.208.37]
Received-SPF: pass (n17.f6.bulk.kks.ynwp.yahoo.co.jp: domain of optinmailpf.gu4tcmrvha3do-okb2qjat=yahoo.co.jp@err.yahoo.co.jp designates 183.79.208.37 as permitted sender) receiver=n17.f6.bulk.kks.ynwp.yahoo.co.jp; client-ip=183.79.208.37; envelope-from=optinmailpf.gu4tcmrvha3do-okb2qjat=yahoo.co.jp@err.yahoo.co.jp;
Authentication-Results: mta086.mail.bbt.yahoo.co.jp  from=mail.yahoo.co.jp; domainkeys=pass (ok); dkim=pass (ok); header.i=@mail.yahoo.co.jp
Received: from 183.79.208.37  (HELO n17.f6.bulk.kks.ynwp.yahoo.co.jp) (183.79.208.37)
  by mta086.mail.bbt.yahoo.co.jp with SMTP; Sun, 29 Apr 2018 15:10:55 +0900
Comments: DKIM? See http://antispam.yahoo.co.jp/efforts_dk.html
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed; d=mail.yahoo.co.jp; s=bulk201106; t=1524982167; bh=HHSiOdFpF5RXFDFh6Z4uRDBFb5hMtf2/ZCj9n4mvJQM=; h=Received:Received:Received:X-yahoo-newman-expires:Date:From:Reply-To:To:Message-ID:Subject:Mime-Version:Content-Type:Content-Transfer-Encoding:X-Yahoo-Newman-Property:X-Yahoo-Newman-Id; b=aUtFA6PRy3H472T0SF41feP7bxboFg49gZV4uhp25RxyL+VI3IDSgq1tRJ9qa5siOXCGFvKL7YnfpJhS6S3pYitoG0IGQSse2tgE1oFziiEB7Jz7+3krHwm3064GGpjsqHmKogROHGAM57cPVQrPKIcW749T4uLvup2syebW3Wc=
DomainKey-Signature: a=rsa-sha1; q=dns; c=nofws; s=bulk201106; d=mail.yahoo.co.jp; 
    b=T6hWpBONMEZhN1mYXXqMI890bdcliIPEML/nptkf1GcTD7y7/qU2c859AnZMfUQsmJjKqB9xK9QlMTzfNt02iZ7PU/b/8lFTVncftxwxSv1l2+bNswmypk+WEy9dF00g/LRndmdUZiFv+a32t9YOgYbJ83gR7qtvMd/Y0N9uqkE=;
Received: from [100.98.128.85] by n17.f6.bulk.kks.ynwp.yahoo.co.jp with NNFMP; 29 Apr 2018 06:09:27 -0000
Received: from [100.97.203.186] by m23.f6.bulk.kks.ynwp.yahoo.co.jp with NNFMP; 29 Apr 2018 06:09:27 -0000
Received: from [127.0.0.1] by worker5184.optin.kks.ynwp.yahoo.co.jp with NNFMP; 29 Apr 2018 06:09:27 -0000
X-yahoo-newman-expires: 1525014000
Date: Sun, 29 Apr 2018 15:09:27 +0900
From: =?UTF-8?B?WWFob28h44Kr44O844OJ?= <yahoocard-master@mail.yahoo.co.jp>
Reply-To: yahoocard-master@mail.yahoo.co.jp
To: okb2qjat@yahoo.co.jp
Message-ID: <5ae56197daf4b_d2041902435710de@worker5184.optin.kks.ynwp.yahoo.co.jp.mail>
Subject: =?UTF-8?Q?=E3=80=90=E6=AF=8E=E9=80=B1=E6=9C=AB72=E6=99=82=E9=96=93=E9=96=8B=E5=82=AC=E3=80=91=E9=80=B1=E6=9C=AB=E3=81=AA=E3=82=89=E6=9C=80=E5=A4=A710,000=E3=83=9D=E3=82=A4=E3=83=B3=E3=83=88=E3=83=97=E3=83=AC=E3=82=BC=E3=83=B3=E3=83=88=EF=BC=81[Yahoo!?=
 =?UTF-8?Q?_JAPAN]?=
Mime-Version: 1.0
Content-Type: text/plain;
 charset=UTF-8
Content-Transfer-Encoding: base64
X-Yahoo-Newman-Id: 7ab92d8c-ab1e-4b77-9227-115988e119fc

CuKUgeKUgeKUgeKUgeKUgeKUgeKUgeKUgeKUgeKUgeKUgeKUgeKUgeKUgeKU
geKUgeKUgeKUgeKUgeKUgeKUgeKUgeKUgeKUgeKUgeKUgeKUgeKUgeKUgeKU
geKUgeKUgQoK44CA44CA44CA44CA44CA44CA44CAWWFob28h44Kr44O844OJ
44CA44GK44GZ44GZ44KB5oOF5aCx44Oh44O844OrCgrilIHilIHilIHilIHi
lIHilIHilIHilIHilIHilIHilIHilIHilIHilIHilIHilIHilIHilIHilIHi
lIHilIHilIHilIHilIHilIHilIHilIHilIHilIHilIHilIHilIEKb2tiMnFq
YXTjgZXjgpPjgavjgYrnn6XjgonjgZvjgZfjgabjgYTjgb7jgZnjgIIKCuOB
hOOBpOOCgllhaG9vISBKQVBBTuOCkuOBlOWIqeeUqOOBhOOBn+OBoOOBjeOB
guOCiuOBjOOBqOOBhuOBlOOBluOBhOOBvuOBmeOAggpZYWhvbyEgSkFQQU7j
gqvjg7zjg4njgYvjgonmlrDopo/lhaXkvJrnibnlhbjjga7jgZTmoYjlhoXj
gafjgZnjgIIKCjo6Ou+8ijo6Ou+8ijo6Ou+8ijo6Ou+8ijo6Ou+8ijo6Ou+8
ijo6Ou+8ijo6Ou+8ijo6Ou+8ijo6OgoKICAgICAgICDilI/ilIHilIHilIHi
lIHilIHilIHilIHilIHilIHilIHilIHilJMKICAgICAgICDilIPvvLzmr47p
gLHmnKs3MuaZgumWk+mWi+WCrO+8jyDilIMKICAgICAgICDilJfilIHilIHi
lIHilIHilIHilIHilIHilIHilIHilIHilIHilJsKCiAg44GE44Gk44KC44Gv
5pyf6ZaT5Zu65a6a77y044Od44Kk44Oz44OIOCwwMDDjg53jgqTjg7Pjg4ji
gLvjgIEK44CA6YCx5pyr44Gq44KJ44GV44KJ44Gr5pyf6ZaT5Zu65a6a77y0
44Od44Kk44Oz44OI77yLMiwwMDDjg53jgqTjg7Pjg4jigLvvvIEKCiDilI/i
lIHilIHilIHilIHilIHilIHilIHilIHilIHilIHilIHilIHilIHilIHilIHi
lIHilIHilIHilIHilIHilIHilIHilIHilIHilIHilIHilIHilJMKIOKUg+KY
heacgOWkpzEwLDAwMOWGhuebuOW9k+OBruacn+mWk+WbuuWumu+8tOODneOC
pOODs+ODiOKAu+ODl+ODrOOCvOODs+ODiOKYhSDilIMKIOKUl+KUgeKUgeKU
geKUgeKUgeKUgeKUgeKUgeKUgeKUgeKUgeKUgeKUgeKUgeKUgeKUgeKUgeKU
geKUgeKUgeKUgeKUgeKUgeKUgeKUgeKUgeKUgeKUmwoKIOKWvOips+OBl+OB
j+OBr+OBk+OBoeOCiQogIGh0dHBzOi8veWFob28uanAvZUFGVFpKCgogICAg
ICAgICAg77yP77yP4pSP4pST77y877y844CA44CAIAogICAgICAgICAgICAg
IOKUo+KUq+OAgOOAgOOAgOOAgCAKICAgICAgICAgICAgICDilIPilKPilLPi
lLPilJPjgIAgCiAgICAgICAgICAgIOKUj+KUgeKUk+KUg+KUg+KUg+OAgAog
ICAgICAgICAgICDilIPilIHilJvilLvilLvilJsKICAgICAgICAgICAg4pSX
4pSTICAgIOKUj+KUm+OAgAoK44CQ6ZaL5YKs5pyf6ZaT77ya5q+O6YCx6YeR
5pucMTI6MDDvvZ7mnIjmm5wxMTo1OeOAkQoKIArjgIDigLvpgLLlkYjjgZnj
govjg53jgqTjg7Pjg4jjga/jgIHmnJ/plpPlm7rlrprvvLTjg53jgqTjg7Pj
g4jjgafjgZnjgIIK44CAWWFob28hIEpBUEFO44CBTE9IQUNP44CBR1lBTyEg
5Lul5aSW44Gu44K144O844OT44K544Gn5Yip55So44Gn44GN44G+44Gb44KT
44CCCuOAgOacieWKueacn+mZkOOBjOOBguOCiuOBvuOBmeOBruOBp+OBlOaz
qOaEj+OBj+OBoOOBleOBhOOAggoK5pyf6ZaT5Zu65a6a77y044Od44Kk44Oz
44OI44Gr44Gk44GE44Gm44Gv44GT44Gh44KJCmh0dHBzOi8veWFob28uanAv
bkE5cUJJCgo6OjrvvIo6OjrvvIo6OjrvvIo6OjrvvIo6OjrvvIo6OjrvvIo6
OjrvvIo6OjrvvIo6OjrvvIo6OjoKCgrilIDilIDilIDilIDilIDilIDilIDi
lIDilIDilIDilIDilIDilIDilIDilIDilIDilIDilIDilIDilIDilIDilIDi
lIDilIDilIDilIDilIDilIDilIDilIDilIDilIAKWWFob28hIEpBUEFO44Kr
44O844OJ44Gv5bm05Lya6LK75rC45bm054Sh5paZ44Gq44Gu44Gn44CBCuOB
iuawl+i7veOBq+OBiueUs+OBl+i+vOOBv+OBhOOBn+OBoOOBkeOBvuOBmeOA
guOBnOOBsuOBiueUs+OBl+i+vOOBv+OBj+OBoOOBleOBhOOAggoK44CA4pa8
WWFob28hIEpBUEFO44Kr44O844OJ44KS55Sz44GX6L6844KA4pa8CuOAgGh0
dHBzOi8veWFob28uanAvZUFGVFpKCgrigLvmnKzjg6Hjg7zjg6vjgajooYzj
gY3pgZXjgYTjgafjgYrnlLPjgZfovrzjgb/jgYTjgZ/jgaDjgYTjgabjgYTj
govloLTlkIjjgavjga/jgZTlrrnotabjgY/jgaDjgZXjgYTjgIIK4pSA4pSA
4pSA4pSA4pSA4pSA4pSA4pSA4pSA4pSA4pSA4pSA4pSA4pSA4pSA4pSA4pSA
4pSA4pSA4pSA4pSA4pSA4pSA4pSA4pSA4pSA4pSA4pSA4pSA4pSA4pSA4pSA
CgpZYWhvbyHjgqvjg7zjg4njgYrjgZnjgZnjgoHmg4XloLHjg6Hjg7zjg6vj
ga/jgIEKWWFob28hIEpBUEFOIElE44Gr44GU55m76Yyy44GV44KM44Gm44GE
44KL44Oh44O844Or44Ki44OJ44Os44K544Gr6YWN5L+h44GX44Gm44GE44G+
44GZ44CCCgrku4rlvozjgZPjga7jg6Hjg7zjg6vjga7phY3kv6HjgpLluIzm
nJvjgZXjgozjgarjgYTmlrnjga/jgIEK44GK5omL5pWw44Gn44GZ44GM5LiL
6YOo44Gr44GC44KL44CM6YWN5L+h44Gu55m76Yyy44O76Kej6Zmk44CN44Gu
VVJM44GL44KJ44CBCllhaG9vISBKQVBBTiBJROOBp+ODreOCsOOCpOODs+OB
l+OBpuaJi+e2muOBjeOCkuOBiumhmOOBhOOBhOOBn+OBl+OBvuOBmeOAggoK
4oC754m55a6a44Gu56uv5pyr44Gn44Gv44CBCuacrOODoeODvOODq+OBruih
qOekuuOChOODquODs+OCr+mBt+enu+OBjOato+OBl+OBj+OBleOCjOOBquOB
hOWgtOWQiOOBjOOBguOCiuOBvuOBmeOAggoK4pahIOeZuuihjO+8muODpOOD
leODvOagquW8j+S8muekvgpodHRwczovL3d3dy55YWhvby5jby5qcC8K4pah
IOS9j+aJgO+8muadseS6rOmDveWNg+S7o+eUsOWMuue0gOWwvuS6leeUujHn
laoz5Y+3CuKWoSDnt6jpm4bvvJpZYWhvbyHjgqvjg7zjg4kKaHR0cHM6Ly9j
YXJkLnlhaG9vLmNvLmpwLwrilqEg55m66KGM5pel77yaMjAxOOW5tDTmnIgy
OeaXpQrilqEg44GK5ZWP44GE5ZCI44KP44GbCmh0dHBzOi8veWFob28uanAv
djAxTVZiCuKWoSDphY3kv6Hjga7nmbvpjLLjg7vop6PpmaQKaHR0cHM6Ly95
YWhvby5qcC84Q1lTVFcK4pahIFlhaG9vISBKQVBBTuOAjOODl+ODqeOCpOOD
kOOCt+ODvOODneODquOCt+ODvOOAjQpodHRwczovL2RvY3MueWFob28uY28u
anAvZG9jcy9pbmZvL3Rlcm1zL2NoYXB0ZXIxLmh0bWwjY2YybmQK4pSA4pSA
4pSA4pSA4pSA4pSA4pSA4pSA4pSA4pSA4pSA4pSA4pSA4pSA4pSA4pSA4pSA
4pSA4pSA4pSA4pSA4pSA4pSA4pSA4pSA4pSA4pSA4pSA4pSA4pSA4pSA4pSA
4pSA4pSA4pSACkNvcHlyaWdodCAoQykgWWFob28gSmFwYW4gQ29ycG9yYXRp
b24uIEFsbCBSaWdodHMgUmVzZXJ2ZWQuIAoK
"

**Comment by devconcept** [comment]
Yep, confirmed, apparently base64 decoding is not working as expected and corrupts data. I got images and html files transferred in base64 being corrupted and not rendering properly.

**Comment by felixhammerl** [comment]
which version are you working with?

**Comment by devconcept** [comment]
I tried the last version published in `npm` just two days ago.

I also found problems when decoding quoted printable messages in `iso-8859-1` charset and isolated the cause in both cases to unnecessary conversions from string to binary and vice-versa.

In the case of base64 the first conversion is located [here](https://github.com/emailjs/emailjs-mime-codec/blob/cb0f03d4078a83f1d1f0cf10a31b8a7e2740a50b/src/mimecodec.js#L86) from a string to a typed array and then to a string in the next line invoked in your code [here](https://github.com/emailjs/emailjs-mime-parser/blob/833c6198ea05b120f9b42813f560e0c7a54cb24a/src/mimeparser.js#L77) and finally to a typed array again [here](https://github.com/emailjs/emailjs-mime-parser/blob/833c6198ea05b120f9b42813f560e0c7a54cb24a/src/mimeparser.js#L351).

In some cases this process fails (not all the time), the first typed array is the correct one and after the conversion to string the data is corrupted and unusable.
